lasagne/layers/recurrent.py

```diff
@@ -0,0 +1,1252 @@
+        applied.
+    hid_init : callable, np.ndarray, theano.shared or TensorVariable
+        Passing in a TensorVariable allows the user to specify
+        the value of `hid_init` (:math:`h_0`). In this mode, `learn_init` is
+        ignored.
+    backwards : bool
+        If True, process the sequence backwards and then reverse the
+        output again such that the output from the layer is always
+        from :math:`x_1` to :math:`x_n`.
+    learn_init : bool
+        If True, initial hidden values are learned. If `hid_init` is a
+        TensorVariable then the TensorVariable is used and
+        `learn_init` is ignored.
+    gradient_steps : int
+        Number of timesteps to include in the backpropagated gradient.
+        If -1, backpropagate through the entire sequence.
+    grad_clipping : False or float
+        If a float is provided, the gradient messages are clipped during the
+        backward pass.  If False, the gradients will not be clipped.  See [1]_
+        (p. 6) for further explanation.
+    unroll_scan : bool
+        If True the recursion is unrolled instead of using scan. For some
+        graphs this gives a significant speed up but it might also consume
+        more memory. When `unroll_scan` is true then the `gradient_steps`
+        setting is ignored.
+    precompute_input : bool
+        If True, precompute input_to_hid before iterating through
+        the sequence. This can result in a speedup at the expense of
+        an increase in memory usage.
+
+    References
+    ----------
+    .. [1] Cho, Kyunghyun, et al: On the properties of neural
+       machine translation: Encoder-decoder approaches.
+       arXiv preprint arXiv:1409.1259 (2014).
+    .. [2] Chung, Junyoung, et al.: Empirical Evaluation of Gated
+       Recurrent Neural Networks on Sequence Modeling.
+       arXiv preprint arXiv:1412.3555 (2014).
+    .. [3] Graves, Alex: "Generating sequences with recurrent neural networks."
+           arXiv preprint arXiv:1308.0850 (2013).
+
+    Notes
+    -----
+    An alternate update for the candidate hidden state is proposed in [2]_:
+
+    .. math::
+        c_t &= \sigma_c(W_{ic} x_t + W_{hc}(r_t \odot h_{t - 1}) + b_c)\\
+
+    We use the formulation from [1]_ because it allows us to do all matrix
+    operations in a single dot product.
+    """
+    def __init__(self, incoming, num_units,
+                 W_in_to_resetgate=init.Normal(0.1),
+                 W_hid_to_resetgate=init.Normal(0.1),
+                 b_resetgate=init.Normal(0.1),
+                 nonlinearity_resetgate=nonlinearities.sigmoid,
+                 W_in_to_updategate=init.Normal(0.1),
+                 W_hid_to_updategate=init.Normal(0.1),
+                 b_updategate=init.Normal(0.1),
+                 nonlinearity_updategate=nonlinearities.sigmoid,
+                 W_in_to_hidden_update=init.Normal(0.1),
+                 W_hid_to_hidden_update=init.Normal(0.1),
+                 b_hidden_update=init.Constant(0.),
```

Comment:
You're right, I didn't notice you had used `Normal`s here, I'll change them to `Constant`s.  Issue here: #53